### ISSUE TITLE
Rule no-system-props does not detect ``Parent.Children`` component tags

### ISSUE THREAD

**Issue by SferaDev** [primer/eslint-plugin-primer-react#26]
Example:

```tsx
             <Header>
                <Header.Item>
                     // Rule does not detect the fontSize
                    <Header.Link href="#" fontSize={2}>
                        // Rule detects the mr
                        <StyledOcticon icon={MarkGithubIcon} size={32} mr={2} />
                        <span>GitHub</span>
                    </Header.Link>
                </Header.Item>
            </Header>
```
